### PR TITLE
ENH: Add cosine-based high-pass filter interface

### LINKED ISSUE #361: Perform simple denoising before estimating CompCor
CompCor can pick on trivial noise patterns that would be removed through other means. To avoid this and let CompCor focus on the non-trivial noise the following simple denoising steps should be performed prior to signal extraction:

- Regressing out non-steady state outliers (see #360)
- High-pass filter with 0.008 Hz (remove signals with slower frequencies than 0.008 Hz)

### PR DESCRIPTION
**Pull request by effigies**
Closes #361.

### CODE REVIEW

**Comment by chrisfilo** [comment]
Great to see this coming along. Some comments:
- We need this for both a and t compcor
- It would be good to avoid saving the bandpass filtered intermediate files
- We should output the cosines in the confounds TSV file

It might be worth considering building this filtering directly into the compcor interface.

File: fmriprep/interfaces/images.py [hunk 1]
```diff
@@ -239,6 +241,49 @@ def _run_interface(self, runtime):
...
+    design_matrix = File(exists=True)
+    residuals = File(exists=True)
+
+
+class HighPassFilter(SimpleInterface):
+    input_spec = HighPassFilterInputSpec
+    output_spec = HighPassFilterOutputSpec
+
+    def _run_interface(self, runtime):
+        TR = 2.5
```
Comment (chrisfilo) [inline fmriprep/interfaces/images.py]: hardcoded?

Comment (effigies) [inline fmriprep/interfaces/images.py]: Temporary.

**Comment by effigies** [comment]
Yeah, if we don't want to have a file of residuals for CompCor to work on, it's going to need to be directly in the interface.

**Comment by effigies** [comment]
Ready for review. Merge depends on nipy/nipype#2107 and bumping niworkflows.

**Comment by chrisfilo** [comment]
Not sure why the tests are failing, but otherwise looks ready to merge.

**Comment by effigies** [comment]
The bug is when the BOLD series is too short to have a cosine regressor. @jdkent resolved this by returning `None`. I don't want to encode that into the nipype interface, so I'll try to figure out a better way to handle that on Wednesday.

**Comment by effigies** [comment]
Anybody know how `mock` works? There's a failure that's [appearing](https://circleci.com/gh/effigies/fmriprep/605#tests/containers/1) in the tests. I can try tracking it down, but perhaps the issue would be obvious to someone who's used it before.

**Comment by effigies** [comment]
The failing test is now passing.

**Pull request merged**